### DISCUSSION TITLE
WebGL not supported in browser

### DISCUSSION THREAD

**Discussion by Yash123Yash123** [vladmandic/face-api#192] [General]
Hi Vladimir, 

How can I use face-api if WebGL is not supported in my browser , is there any other way to use face-api in browser ?

Getting this message "The highest priority backend 'wasm' has not yet been initialized. Make sure to await tf.ready() or await tf.setBackend() before calling other methods"

Thanks

**Comment by vladmandic** [comment]
what you can do is use version of face-api without bundled tfjs and load tfjs backend you want to use explicitly.  
something like that is noted in readme as a nodejs example, just convert it to browser syntax and use `dist/face-api.esm-nobundle.js`

```js
  const tf = require('@tensorflow/tfjs');
  const wasm = require('@tensorflow/tfjs-backend-wasm');
  const faceapi = require('@vladmandic/face-api/dist/face-api.node-wasm.js'); // use this when using face-api in dev mode
  wasm.setWasmPaths('https://cdn.jsdelivr.net/npm/@tensorflow/tfjs-backend-wasm/dist/');
  await tf.setBackend('wasm');
  await tf.ready();
```
